form/data_mappers.rst

```diff
@@ -177,19 +181,41 @@ method to configure the data mapper::
             // when creating a new color, the initial data should be null
             $resolver->setDefault('empty_data', null);
         }
+
+        // ...
     }
 
-Cool! When using the ``ColorType`` form, the custom ``ColorMapper`` will create
-a new ``Color`` object now.
+Cool! When using the ``ColorType`` form, the custom data mapper methods will
+create a new ``Color`` object now.
 
 .. caution::
 
     When a form has the ``inherit_data`` option set to ``true``, it does not use the data mapper and
     lets its parent map inner values.
 
-.. tip::
+.. sidebar:: Statefull Data Mappers
+
+    Sometimes, data mappers need to access values from a container or contain a
+    state. In this case, you cannot implement the methods in the form type
+    itself. Create a separate class implementing ``DataMapperInterface`` and
+    initialize it in your form type::
 
-    You can also implement the ``DataMapperInterface`` in the ``ColorType`` and add
-    the ``mapDataToForms()`` and ``mapFormsToData()`` in the form type directly
-    to avoid creating a new class. You'll then have to call
-    ``$builder->setDataMapper($this)``.
+        // src/AppBundle/Form/Type/ColorType.php
+
+        // ...
+        use AppBundle\Form\DataMapper\ColorMapper;
+
+        final class ColorType extends AbstractType
+        {
+            public function buildForm(FormBuilderInterface $builder, array $options)
+            {
+                $builder
+                    // ...
+
+                    // Initialize the data mapper class and i.e. pass some state
```

Comment:
i.e. = in other words
e.g. = for example

So I think here we should use "e.g."